src/triage/component/catwalk/model_testers.py

```diff
@@ -0,0 +1,83 @@
+import logging
+
+
+from triage.component.catwalk.predictors import Predictor
+from triage.component.catwalk.individual_importance import IndividualImportanceCalculator
+from triage.component.catwalk.evaluation import ModelEvaluator
+
+
+class ModelTester(object):
```

Comment:
I don't like lodging a critique or making a complaint without having a suggestion, but I'm not sure I'm in the right place to recommend a better or more appropriate name than `ModelTester`. That said, the word "test" is unexpected in this context, (though I entirely understand that you are "testing" a model, rather than testing software or something).

…Maybe: `ModelTrial`?

Reply:
These specifically are operations on a given test matrix (the 'test matrix' nomenclature isn't going anywhere, I think). The thing that really binds these three operations together is that test matrices can be giant and you often want to load them into memory once, do everything you need to do on them, and then be done with them. Splitting these up into discrete tasks can waste a lot of I/O, and these problems are magnified in a multiprocessing scenario.

My problem with ModelTrial is that in the stats world 'trials' mean something specific too. At any rate, including 'model' in the name was probably an error on my part because it operates on multiple models. Maybe the name should invoke 'Matrix' some more.

This also makes me think that the train prediction and train evaluation could be moved out of here. They are bundled here for code convenience, but perhaps they should be bundled with other train matrix operations.

Reply:
I don't have a better thing here. I have not had luck inventing names for things for data scientists in Triage that make more sense to engineers. They seem to use 'test' for this stuff, and don't really ever test software so that's not on their mind to conflict with this.

The best I can argue is that since Model is right there in the name, it should be enough to disambiguate from software testing concerns, and that's really the best we can do for now.

Reply:
:+1: makes sense to me